src/rust/engine/options/src/lib.rs

```diff
@@ -257,8 +258,23 @@ impl OptionParser {
             include_derivation: false,
         };
 
-        fn path_join(a: &str, b: &str) -> String {
-            format!("{}{}{}", a, path::MAIN_SEPARATOR, b)
+        fn path_join(prefix: &str, suffix: &str) -> String {
+            if prefix.ends_with(path::MAIN_SEPARATOR) {
+                format!("{}{}", prefix, suffix)
+            } else {
+                format!("{}{}{}", prefix, path::MAIN_SEPARATOR, suffix)
+            }
+        }
+
+        fn path_strip(prefix: &str, path: &str) -> String {
```

Comment:
For paths under the buildroot, it seemed more useful to put the relative path to the buildroot in the derivation, rather than the full path. It also makes the tests simpler. 

Note that we implement our own simply path manipulation here because we have Strings in hand, not OsString or Path.

Reply:
The problem is going back to a String from the underlying OsString, but for now I've done it with an aggressive unwrap(), and in the future the calling code should really be using OsString or Path anyway.